src/main/java/io/gravitee/repository/cassandra/management/CassandraEventRepository.java

```diff
@@ -0,0 +1,282 @@
+                .and(set("parent_id", event.getParentId()))
+                .and(set("properties_api_id", event.getProperties().get("api_id")))
+                .and(set("properties_origin", event.getProperties().get("origin")))
+                .and(set("properties_username", event.getProperties().get("username")))
+                .and(set("updated_at", event.getUpdatedAt()))
+                .where(eq("id", event.getId()));
+
+        Statement updateType = QueryBuilder.update(EVENTS_TYPE_TABLE)
+                .with(set("type", event.getType().toString()))
+                .where(eq("id", event.getId()));
+
+        Statement updateProperties = QueryBuilder.update(EVENTS_PROPERTIES_TABLE)
+                .with(set("properties_api_id", event.getProperties().get("api_id")))
+                .and(set("properties_origin", event.getProperties().get("origin")))
+                .and(set("properties_username", event.getProperties().get("username")))
+                .where(eq("id", event.getId()));
+
+        Statement updateDate = QueryBuilder.update(EVENTS_DATE_TABLE)
+                .with(set("updated_at", event.getUpdatedAt()))
+                .where(eq("id", event.getId()));
+
+        BatchStatement batchStatement = new BatchStatement();
+        batchStatement.add(update).add(updateType).add(updateProperties).add(updateDate);
+        session.execute(batchStatement);
+
+        final Optional<Event> updatedEvent = findById(event.getId());
+
+        if (updatedEvent.isPresent()) {
+            LOGGER.debug("Event updated");
+            return updatedEvent.get();
+        }
+
+        return null;
+    }
+
+    @Override
+    public void delete(String eventId) throws TechnicalException {
+        LOGGER.debug("Delete Event [{}]", eventId);
+
+        Statement delete = QueryBuilder.delete().from(EVENTS_TABLE).where(eq("id", eventId));
+        Statement deleteType = QueryBuilder.delete().from(EVENTS_TYPE_TABLE).where(eq("id", eventId));
+        Statement deleteProperties = QueryBuilder.delete().from(EVENTS_PROPERTIES_TABLE).where(eq("id", eventId));
+        Statement deleteDate = QueryBuilder.delete().from(EVENTS_DATE_TABLE).where(eq("id", eventId));
+
+        BatchStatement batchStatement = new BatchStatement();
+        batchStatement.add(delete).add(deleteType).add(deleteProperties).add(deleteDate);
+        session.execute(batchStatement);
+
+    }
+
+    private List<Event> searchEvent(EventCriteria filter) {
+        LOGGER.debug("Entering search()");
+
+        // Get event ids by type of event
+        List<String> idsType = null;
+        List<String> types = filter.getTypes().stream().map(Enum::toString).collect(Collectors.toList());
+        if (types != null && !types.isEmpty()) {
+            idsType = new ArrayList<>();
+            for (String type: types) {
+                Statement st = QueryBuilder.select("id").from(EVENTS_TYPE_TABLE).allowFiltering().where(eq("type", type));
```

Comment:
Using **eq** avoids querying across partitions and GC staggering.
With **in**, queries & responses are kept in memory while all partitions are requested. It is fine when few requests are made, but it can become problematic over time.
[source](http://www.datastax.com/dev/blog/a-deep-look-to-the-cql-where-clause)